### PR TITLE
Add Additional Search Fields and Search Highlighting

### PR DESCRIPTION
**Pull request by mstruve**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
While hooking up a lot of the front end code I found some fields that we were missing in Elasticsearch so I added them. In addition, we use highlighting in the main feed. For example if you search "Ruby" it will show you the results of your search along with highlighted snippets from the article where "Ruby" was found. This adds that functionality to our searching and updates our docs to return it. 

Notice, a lot of the fields I am adding are purely for display purposes. _Ideally_, you don't want to have fields in Elasticsearch unless you are searching and using them. My hope is that we will end up searching and using these fields in the future. If not, then we can revisit removing them. For now, I think getting us flipped to Elasticsearch is worth putting an additional display field or 5(😂 ) into our documents. 

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/projects/6#card-34785956

## Added tests?
- [x] yes

## Added to documentation?
I added comments!

alt_text

### CODE REVIEW

File: app/controllers/search_controller.rb [hunk 1]
```diff
@@ -25,6 +25,7 @@ class SearchController < ApplicationController
     sort_by
     tag_names
     user_id
+    class_name
```
Comment (mstruve) [inline app/controllers/search_controller.rb]: Need to permit this for searching

File: app/services/search/query_builders/feed_content.rb [hunk 2]
```diff
@@ -62,7 +86,7 @@ def term_keys
         TERM_KEYS.map do |term_key, search_key|
           next unless @params.key? term_key
 
-          { term: { search_key => @params[term_key] } }
+          { terms: { search_key => Array.wrap(@params[term_key]) } }
```
Comment (mstruve) [inline app/services/search/query_builders/feed_content.rb]: `term` queries are for searching single values, [`terms` queries](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-query.html#query-dsl-terms-query) are what you use when you want to search an array of values. The default search between items in the array is `OR`.

File: spec/services/search/feed_content_spec.rb [hunk 3]
```diff
@@ -69,7 +81,7 @@
 
       it "filters by class_name" do
         pde = create(:podcast_episode)
-        index_documents([pde, article1, article2])
+        index_documents([article1, article2, pde])
```
Comment (mstruve) [inline spec/services/search/feed_content_spec.rb]: unintended change from an attempted spec fix that is not harmful so I left it.

File: spec/services/search/feed_content_spec.rb [hunk 4]
```diff
@@ -18,6 +18,18 @@
       expect(described_class).to have_received(:search).with(body: a_kind_of(Hash))
     end
 
+    it "returns highlighted fields" do
+      allow(article1).to receive(:body_text).and_return("I love ruby")
+      allow(article2).to receive(:body_text).and_return("Ruby Tuesday is yummy")
```
Comment (atsmith813) [inline spec/services/search/feed_content_spec.rb]: 😂

File: app/models/podcast_episode.rb [hunk 5]
```diff
@@ -75,6 +75,10 @@ def user_name
     podcast_title
   end
 
+  def podcast_image
+    ProfileImage.new(podcast).get(width: 90)
+  end
```
Comment (rhymes) [inline app/models/podcast_episode.rb]: as this is only called in the serializer, would it make sense to keep it there in a block?

**Pull request merged**